e2e_tests/tests/conftest.py

```diff
@@ -220,3 +220,8 @@ def collect_trial_profiles(record_property: Callable[[str, object], None]) -> Ca
     """
 
     return record_profiling.profile_test(record_property=record_property)
+
+
+@pytest.fixture(scope="session")
```

Comment:
The docs are not very clear about it, but the reason their example shows a similar decorator is to generate a temp directory with session scope (scoped to the whole pytest execution). As mentioned in another comment, when you want a temp directory that's just for each individual test (like the tests here), you can just use the bulit-in fixtures.

Reply:
Got it, thanks!